### PR TITLE
feat(explore): metrics and filters controls redesign

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY

Redesign Metrics and Filters controls according to design in https://www.figma.com/file/JWaGztdhZS0kS5ruG7x9tB/Control-Panel?node-id=195%3A36068

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
image

After: 
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC: @villebro @junlincc

### CODE REVIEW

File: superset-frontend/src/explore/components/controls/MetricsControl.jsx
Comment (villebro) [inline superset-frontend/src/explore/components/controls/MetricsControl.jsx:139]: I wonder if we could still leverage this logic for setting the default aggregate if it's unset and the user chooses the column first (the usual case)?

**Review by villebro — COMMENTED** [review]
This is looking AWESOME! 🎉  Some first pass comments. I encountered numerous bugs when testing this that turned out to be old bugs. I'll be opening a PR to address some of those in a second.

**Comment by junlincc** [comment]
ezgif-6-5a4e83231ee2

@kgabryje much better UI!! many thanks for making it happen!

**Comment by villebro** [comment]
@kgabryje during testing I noticed some strange behavior, some of which was actually present from before, like inconsistent conversion of simple filters to sql filters and labels and how the `isValid()` function deals with missing values, causing false positives. I have a follow up PR that's mostly done which makes more sense to open after this is merged. So I suggest merging this when you feel this is ready and has had some more review and I'll follow up with a PR to fix some of those inconsistencies.

**Comment by ktmud** [comment]
This looks so nice. Can't wait to get rid of react-select!

**Review by junlincc — APPROVED** [review]

**Comment by mihir174** [comment]
Hey all, here are design revisions that I just discussed with @villebro and @junlincc to allow users to add saved metrics that are defined in the dataset - https://www.figma.com/file/JWaGztdhZS0kS5ruG7x9tB/Control-Panel?node-id=195%3A36068

Screen Shot 2020-12-17 at 1 56 57 PM

Since there are 3 ways to add a metric, so it would be confusing if 2 or more of these ways were accessed from the same set of controls. Having one tab per way of adding seems like the most intuitive solution.
@kgabryje

**Comment by junlincc** [comment]
we are temporarily losing the saved metrics which used to show in the Select Column dropdown. we will bring those back by implementing above design @mihir174 proposed ^^^^^  ASAP in a separate PR in the following week. 

cc: @ktmud @graceguo-supercat @etr2460

**Review by villebro — APPROVED** [review]
As this is a major step on the SIP-34 roadmap, we're going to merge this to unblock other PRs that are dependent on this functionality. There are some known limitations in this implementation, but these will be addressed (and improved upon) in follow up PRs that are set to arrive shortly.

**Comment by ktmud** [comment]
👏

**Comment by ktmud** [comment]
Hi, finally got a chance to test this locally.  Some feedbacks on the UX:

1. The focus state outline highlight got cut off:
   
2. The metric pills for pre-defined metrics should show metric definition, either in a info icon tooltip or when clicked:
   
3.  @mihir174 what do you think of removing the "+" icon but always show the "Add metric" pill that is currently only used in the empty state.
4. Can we please add drag & drop to sort the metrics soon? 


Not sure if these are "known limitations", but I hope it got addressed soon. since this is not behind a feature flag, we basically can't deploy if the feature is incomplete (Good thing is we don't have to ship for at least another two weeks!)

**Comment by kgabryje** [comment]
Thank you for your comments @ktmud, I'll work on those

**Comment by junlincc** [comment]
1,2,4 yes. 3 not yet until @mihir174 gets back. 🙏

**Comment by zhaoyongjie** [comment]
(5) can be done together with drag and drop support in Explore.
@ktmud

**Comment by ktmud** [comment]
Added some clarification for (5). This is basically about having full feature parity of #9627 which implements #7129.

**Comment by junlincc** [comment]
got it, i misunderstood. will get to it.

**Comment by ktmud** [comment]
> @mihir174 what do you think of removing the "+" icon but always show the "Add metric" pill that is currently only used in the empty state. It's a little weird that two CTA living so close are basically doing the same thing.

Bump this one up.  We've had user report that they didn't know how to add a second metric. So this seems to be another side-effect of two CTAs.

cc @junlincc @etr2460

**Pull request merged**
